admin/class-yoast-form.php

```diff
@@ -863,8 +863,8 @@ public function index_switch( $var, $label, $help = '', $attr = [] ) {
 			$index_switch_values,
 			sprintf(
 				/* translators: %s expands to an indexable object's name, like a post type or taxonomy */
-				esc_html__( 'Show %s in search results?', 'wordpress-seo' ),
-				esc_html( $label )
+				__( 'Show %s in search results?', 'wordpress-seo' ),
+				$label
```

Comment:
You should try to escape it on following way:
`'<a href="' . esc_url( $archive_url ) . '">' . esc_html( $label ) .'</a>'`